### ISSUE TITLE
[NativeAOT/osx-arm64] Warnings publishing classlib with `StripSymbols=true`

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtime#80934]
https://github.com/dotnet/runtime/pull/80515#issuecomment-1385740559

## Repro

Publish classlib on osx-arm64 with `PublishAot=true` and `StripSymbols=true`

## Result
```
  Generating native code
EXEC : warning : (arm64)  could not find object file symbol for symbol __Z19InitializeCpuCGroupv [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __Z16CleanupCpuCGroupv [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __Z11GetCpuLimitPj [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __ZN6CGroup14FindCGroupPathEPFbPKcE [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __ZN6CGroup21IsCGroup1CpuSubsystemEPKc [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __ZN6CGroup18FindHierarchyMountEPFbPKcEPPcS5_ [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __ZN6CGroup26FindCGroupPathForSubsystemEPFbPKcE [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __ZN6CGroup18GetCGroup1CpuLimitEPj [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __ZN6CGroup18GetCGroup2CpuLimitEPj [/Users/jkotas/xxx/xxx.csproj]
EXEC : warning : (arm64)  could not find object file symbol for symbol __ZN6CGroup25ReadLongLongValueFromFileEPKcPx [/Users/jkotas/xxx/xxx.csproj]
  xxx -> /Users/jkotas/xxx/bin/Release/net8.0/osx-arm64/publish/
```

**Comment by jkotas** [comment]
@am11 Could you please take a look?

**Comment by am11** [comment]
Sure, I am waiting for the next daily build of installer to test it end-to-end.

TL;DR this issue is limited to the local builds.

In the latest daily build of SDK (`8.0.100-alpha.1.23070.23`, runtime commit: ddb6988968), there were no missing symbols warnings. However, when I built ddb6988968 from source (`./build.sh clr+libs+packs -c Release`) and tested the local package:

```xml
<!-- nuget.config -->
<add key="foo" value="/Users/am11/projects/runtime/artifacts/packages/Release/Shipping" /> 
```

```csproj
<!-- csproj -->
<PackageReference Include="Microsoft.DotNet.ILCompiler" Version="8.0.0-dev" />
<!-- or -->
<PackageReference Include="Microsoft.DotNet.ILCompiler; runtime.osx-arm64.Microsoft.DotNet.ILCompiler"
  Version="8.0.0-dev" />
```

the warning re-appeared. I cleared the nuget caches and obj/bin directories after each step but this difference persists.

```sh
% rm -rf obj bin dist pkg
% dotnet8 publish -p:PublishAot=true -p:StripSymbols=true -c Release -o dist --ucr --packages pkg --no-cache
```

I haven't figured out the difference between local build and daily build on official machine, but after some more testing with a few more nearby versions one thing is becoming clear that this difference exists on osx-arm64. Could be the tools versions or local nuget consumption being off?

Other info:
```sh
% xcodebuild -version 
Xcode 14.2
Build version 14C18

% cmake --version
cmake version 3.24.1

# macos version
% sw_vers -productVersion 
13.1
```

Also, I did a quick test with PackageReference with code-flow versions (`8.0.0-alpha.1.23070.6` from https://github.com/dotnet/sdk/pull/30065/files that includes #80515) and there are no warnings.

**Comment by am11** [comment]
I tested with today's installer build `8.0.100-alpha.1.23073.15` (runtime: 7ec3634ee5) and it doesn't give the warnings on M1.
